pbft/pbft.cpp

```diff
@@ -83,6 +85,23 @@ pbft::start()
                 this->audit_heartbeat_timer->async_wait(
                         std::bind(&pbft::handle_audit_heartbeat_timeout, shared_from_this(), std::placeholders::_1));
 
+                this->service->register_execute_handler(
+                        [weak_this = this->weak_from_this(), fd = this->failure_detector]
+                                (const pbft_request& req, uint64_t sequence)
+                                        {
+                                            fd->request_executed(req);
+
+                                            if (sequence % CHECKPOINT_INTERVAL == 0)
```

Comment:
Is there an error if weak_this is not around anymore? 
Should there be logging at least?